paimon-core/src/main/java/org/apache/paimon/io/DataFileMeta.java

```diff
@@ -120,6 +121,13 @@ public class DataFileMeta {
 
     private final @Nullable List<String> valueStatsCols;
 
+    /**
+     * the data root location that the file resides in, if it is null, the file is in the default
+     * warehouse path, when {@link CoreOptions#DATA_FILE_PATH_DIRECTORY} is set, new writen files
+     * will be persisted in {@link CoreOptions#DATA_FILE_PATH_DIRECTORY}.
+     */
+    private final @Nullable String dataRootLocation;
```

Comment:
We need to introduce a new version for CommitMessage and DataSplit.

Reply:
I see that since version 0.9 [1], the versions of commitMessage and DataSplit have been changed. Do I need to make another change? I think it only needs to be changed once in version 1.0.

[1]  https://github.com/apache/paimon/blob/release-0.9/paimon-core/src/main/java/org/apache/paimon/table/source/DataSplit.java
[1]  https://github.com/apache/paimon/blob/release-0.9/paimon-core/src/main/java/org/apache/paimon/table/sink/CommitMessageSerializer.java

Reply:
Keep compatibility, even it is just in 1.0-SNAPSHOT, for example, creating `DataFileMeta10LegacySerializer` for previous DataFileMetaSerializer.

Reply:
Understood, if it's for compatibility with 1.0-SNAPSHOT, it's worth doing this.